typing/parmatch.ml

```diff
@@ -2006,16 +2006,22 @@ let irrefutable pat = le_pat pat omega
 
 let rec inactive pat =
   match pat.pat_desc with
-  | Tpat_lazy _ ->
+  | Tpat_lazy _ | Tpat_array _ ->
       false
-  | Tpat_any | Tpat_var _ | Tpat_constant _ | Tpat_variant (_, None, _) ->
+  | Tpat_any | Tpat_var _ | Tpat_variant (_, None, _) ->
       true
-  | Tpat_tuple ps | Tpat_construct (_, _, ps) | Tpat_array ps ->
+  | Tpat_constant c -> begin
+      match c with
+      | Const_string _ -> false
```

Comment:
How would you mutate a string literal used in a constant pattern?

Reply:
You could imagine:

```ocaml
let f ("foo" as c) = fun () -> c;;
let s = "foo";;
let g = f "foo";;
s.[0] <- 'g';;
g ()
```

Although in this case the functions won't be merged as the match is not (and cannot be) exhaustive.
Still, the code "feels more correct".

Reply:
Ah, right, sorry, the problem is of course mutating the matched value, not the pattern.

Could you adapt to take into account Config.safe_string?  (I understand this is a bit useless since string literal patterns are not exhaustive.)

Reply:
(I have a feeling that what I wrote makes no sense at all, oh well...)